### PR TITLE
feat: add alpine.auth.jwt.ttl.seconds config

### LINKED ISSUE #534: Add dynamic JWT Token expiry time
Currently, the expiration time for JWT tokens is hardcoded to 7 days in the code. To enhance security measures, it is recommended to allow specifying the expiration time in minutes or hours dynamically.

```
public String createToken(final Principal principal, final List<Permission> permissions, final IdentityProvider identityProvider) {
    final Date today = new Date();
    final JwtBuilder jwtBuilder = Jwts.builder();
    jwtBuilder.setSubject(principal.getName());
    jwtBuilder.setIssuer(ISSUER);
    jwtBuilder.setIssuedAt(today);
    jwtBuilder.setExpiration(addDays(today, 7)); // Hardcoded expiration time

    // ... (additional code)

    // Request: Allow dynamic setting of token expiration time (in minutes or hours)
}
```
Please consider implementing this enhancement for better flexibility and security in managing JWT token expiration times. Feel free to reach out if further clarification.

Feel free to add specific implementation approaches in the comments, and I will try to create a PR  by myself as soon as possible.

Best
DoPaNik

### PR DESCRIPTION
**Pull request by ShuP1**
Fix #534

Should be add to dependency-track doc after the merge

### CODE REVIEW

**Comment by stevespringett** [comment]
@nscuro thoughts?

File: alpine-common/src/main/java/alpine/Config.java [hunk 1]
```diff
@@ -181,7 +181,8 @@ public enum AlpineKey implements Key {
         CORS_ALLOW_CREDENTIALS                 ("alpine.cors.allow.credentials",     true),
         CORS_MAX_AGE                           ("alpine.cors.max.age",               3600),
         WATCHDOG_LOGGING_INTERVAL              ("alpine.watchdog.logging.interval",  0),
-        API_KEY_PREFIX                         ("alpine.api.key.prefix",             "alpine_");
+        API_KEY_PREFIX                         ("alpine.api.key.prefix",             "alpine_"),
+        AUTH_JWT_TTL                           ("alpine.auth.jwt.ttl",               7 * 24 * 60);
```
Comment (nscuro) [inline alpine-common/src/main/java/alpine/Config.java:185]: Consider naming it `alpine.auth.jwt.ttl.seconds` to make it less ambiguous unit-wise.

File: alpine-server/src/main/java/alpine/server/auth/JsonWebToken.java [hunk 2]
```diff
@@ -130,13 +130,30 @@ public String createToken(final Principal principal, final List<Permission> perm
...
+    /**
+     * Creates a new JWT for the specified principal. Token is signed using
+     * the SecretKey with an HMAC 256 algorithm.
+     *
+     * @param principal the Principal to create the token for
+     * @param permissions the effective list of permissions for the principal
+     * @param identityProvider the identity provider the principal was authenticated with. If null, it will be derived from principal
+     * @param ttlSeconds the token time-to-live in seconds
+     * @return a String representation of the generated token
+     * @since 2.2.5
```
Comment (nscuro) [inline alpine-server/src/main/java/alpine/server/auth/JsonWebToken.java:148]: Should be changed to 2.2.6 now.

File: alpine-server/src/main/java/alpine/server/auth/JsonWebToken.java [hunk 3]
```diff
@@ -130,13 +130,30 @@ public String createToken(final Principal principal, final List<Permission> perm
...
+     */
+    public String createToken(final Principal principal, final List<Permission> permissions, final IdentityProvider identityProvider, final int ttlSeconds) {
         final Date today = new Date();
```
Comment (nscuro) [inline alpine-server/src/main/java/alpine/server/auth/JsonWebToken.java:151]: Consider renaming `today` to `now` so it matches the `addDays` -> `addSeconds` change.

**Review by nscuro — CHANGES_REQUESTED** [review]
Overall looks good to me, just a few minor suggestions.

File: alpine-common/src/main/java/alpine/Config.java (hunk 1 above)
Comment (ShuP1) [reply alpine-common/src/main/java/alpine/Config.java:185]: Should I rename `AUTH_JWT_TTL` too ?

Comment (nscuro) [reply alpine-common/src/main/java/alpine/Config.java]: Yes please.

**Review by nscuro — APPROVED** [review]

**Pull request merged**